src/shared/misc.c

```diff
@@ -546,24 +548,48 @@ pid_t get_pid(const char *applet,const char *pidfile)
 struct ready ready_parse(const char *applet, const char *ready_string)
 {
 	struct ready ready = {0};
-	if (sscanf(ready_string, "fd:%d", &ready.fd) != 1)
+	if (sscanf(ready_string, "fd:%d", &ready.fd) == 1) {
+		ready.type = READY_FD;
+		if (pipe(ready.pipe) == -1)
+			eerrorx("%s: pipe: %s", applet, strerror(errno));
+	} else if (strcmp(ready_string, "socket") == 0) {
+		union {
+			struct sockaddr header;
+			struct sockaddr_un unix;
+		} addr = { .unix = { .sun_family = AF_UNIX } };
+		int written;
+
+		ready.type = READY_SOCKET;
+		written = snprintf(addr.unix.sun_path, sizeof(addr.unix.sun_path), "%s/supervise-%s.sock", rc_svcdir(), applet);
+		if (written > (int) sizeof(addr.unix.sun_path))
+			eerrorx("%s: socket path '%s/supervise-%s.sock' too long.", applet, rc_svcdir(), applet);
+		else if (written < 0)
+			eerrorx("%s: snprintf failed.", applet);
```

Comment:
I don't think snprintf can ever fail, unless you give it invalid format specifier or length argument that's bigger than `INT_MAX`. Neither of which happens here.